contentcuration/contentcuration/frontend/channelEdit/vuex/contentNode/actions.js

```diff
@@ -5,14 +5,50 @@ import client from 'shared/client';
 import { RELATIVE_TREE_POSITIONS } from 'shared/data/constants';
 import { ContentNode, Tree } from 'shared/data/resources';
 import { promiseChunk } from 'shared/utils';
-
-export function loadContentNodes(context, params = {}) {
-  return ContentNode.where(params).then(contentNodes => {
-    context.commit('ADD_CONTENTNODES', contentNodes);
-    return contentNodes;
-  });
+import throttle from 'lodash/throttle';
+import memoize from 'lodash/memoize';
+
+// lodash.memoize uses the first argument passed to the memoized fn
+// as its cache key. The first arg is always vuex context, so every call
+// may just use the same memoized results despite receiving different
+// parameters. We splice(1) to remove context altogether.
+const memoCacheResolver = (...args) => JSON.stringify(args.splice(1));
```

Comment:
Note that this cache resolver may result in cache misses, due to differing key orders in the parameters object.

To fix this, we either need to be specific about which parameters we want to accept and generate a key based on that, or we need to order the keys of the params object before stringifying.

See here for an example:
https://github.com/learningequality/kolibri/blob/release-v0.14.x/kolibri/core/assets/src/api-resource.js#L607

Reply:
Ooooh great catch. I implemented a fn to do this recursively for nested objects as well.